kore/test/Test/Kore/AllPath.hs

```diff
@@ -0,0 +1,347 @@
+module Test.Kore.AllPath where
+
+import Test.Tasty
+
+import           Control.Applicative
+import qualified Data.Foldable as Foldable
+import           Data.Function
+                 ( (&) )
+import           Data.Functor.Identity
+import qualified Data.Graph.Inductive as Gr
+import qualified Data.Maybe as Maybe
+import           Data.Sequence
+                 ( Seq )
+import qualified Data.Sequence as Seq
+import           GHC.Stack
+                 ( HasCallStack )
+
+import qualified Kore.AllPath as AllPath
+import qualified Kore.Step.Representation.MultiOr as MultiOr
+import qualified Kore.Step.Strategy as Strategy
+import           Kore.Step.Transition
+                 ( runTransitionT )
+import qualified Kore.Step.Transition as Transition
+
+import Test.Kore.Comparators ()
+import Test.Tasty.HUnit.Extensions
+import Test.Terse
+
+type ExecutionGraph = Strategy.ExecutionGraph (AllPath.ProofState Integer) ()
+
+emptyExecutionGraph :: AllPath.ProofState Integer -> ExecutionGraph
+emptyExecutionGraph = Strategy.emptyExecutionGraph
+
+insNode
+    :: (Gr.Node, AllPath.ProofState Integer)
+    -> ExecutionGraph
+    -> ExecutionGraph
+insNode = Strategy.insNode
+
+insEdge
+    :: (Gr.Node, Gr.Node)
+    -> ExecutionGraph
+    -> ExecutionGraph
+insEdge = Strategy.insEdge
+
+test_unprovenNodes :: [TestTree]
+test_unprovenNodes =
+    [ AllPath.unprovenNodes
+        (emptyExecutionGraph AllPath.Proven)
+        `satisfies_`
+        Foldable.null
+    , AllPath.unprovenNodes
+        (emptyExecutionGraph (AllPath.Goal 1))
+        `satisfies_`
+        (not . Foldable.null)
+    , AllPath.unprovenNodes
+        (emptyExecutionGraph (AllPath.Goal 1))
+        `equals`
+        (MultiOr.MultiOr [1])
+        $  "returns single unproven node"
+    , AllPath.unprovenNodes
+        (emptyExecutionGraph (AllPath.Goal 0)
+            & insNode (1, AllPath.Goal 1)
+            & insNode (2, AllPath.Proven)
+        )
```

Comment:
Is such a case possible?

Reply:
Such an `ExecutionGraph` would not arise from executing any of the strategies we employ, but I would like to make sure that `AllPath.unprovenNodes` behaves nicely over any graph to avoid dynamic entanglements.